packages/api/tests/lambdas/create-reconciliation-report.js

```diff
@@ -101,27 +102,25 @@ function storeCollectionsToDynamoDb(tableName, collections) {
   return storeToDynamoDb(tableName, putRequests);
 }
 
-async function fetchCompletedReport(Bucket, stackName) {
-  const Prefix = `${stackName}/reconciliation-reports/`;
-
-  const report = await awsServices.s3().listObjectsV2({ Bucket, Prefix }).promise()
-    .then((response) => response.Contents[0].Key)
-    .then((Key) => awsServices.s3().getObject({ Bucket, Key }).promise())
-    .then((response) => response.Body.toString())
-    .then(JSON.parse);
-
-  if (report.status === 'RUNNING') {
+async function fetchCompletedReport(t) {
+  const reportRecord = await new models.ReconciliationReport().scan();
+  t.is(reportRecord.Items.length, 1);
+  if (reportRecord.Items[0].status === 'Pending') {
```

Comment:
When would be in a situation where the status is Pending? Doesn't the handler wait for it to finish before returning? We probably need to move rec report generation to be an async operation, but not for this PR. Something to keep in mind.

Reply:
I updated, the lambda should wait.  It's waiting for the s3 file get updated.

Reply:
Actually I don't understand the original logic, s3 reading should be eventually consistent. I removed it.